integrationtest/mapper/pojo-base/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/definition/ProjectionConstructorFieldProjectionIT.java

```diff
@@ -134,8 +134,10 @@ class IndexedEntity {
 		class MyProjection {
 			public final String myEnum;
 
+			@SuppressWarnings("deprecation")
 			@ProjectionConstructor
-			public MyProjection(@FieldProjection(convert = ValueConvert.NO) String myEnum) {
+			public MyProjection(
+					@FieldProjection(valueModel = ValueModel.INDEX) String myEnum) {
```

Comment:
That's nice but shouldn't `ValueConvert` be tested as well?

Reply:
that's related to your comment above: https://github.com/hibernate/hibernate-search/pull/4214#discussion_r1660706416

I've removed the value convert so that's why the test looks like that.... 

Now, going back to that comment... I've started with `@FieldProjection` having two attributes, one existing for value convert and the other for a new enum. The problem with that is handling two values... I had some code in the processor that would check that both value convert and new enum values are not conflicting. But that means that the user has to set both values at the same time... because if the user changes one value but keep the other default ... what should we pick... so IDK but it seems that just removing the attribute makes it cleaner... an alternative I could think of would be to create a new `FieldProjection` annotation and deprecate existing one, but then how to call the new annotation ... 😖😖😖😖😖😖😖😖😖 😕

Reply:
Maybe do what you did in your first attempt, but ignore `convert` if it's set to the default `YES`? Not great, but at least it sort of works and detects most misuses -- with the remaining ones having a warning.

Reply:
Ok, so I've added back the convert attribute, but it still looks a bit useless to me there...
If the convert value is not a default one i.e. `convert != YES` it means that there's a conflict between `convert` and a default `modelValue` (which is `MAPPING`). If the model value is set to non-default `INDEX` to match `convert=NO` then it means that the user went in and made a change to the annotation ... so to me, it seems if you are going to apply the change anyway, then why not drop the deprecated attribute and use the new one... and if the user does that then it is the same as if we just drop the `convert` attribute entirely... 
Logging a warning instead of throwing an exception, in this case, seems like something that would lead to problems at runtime.

Reply:
Oh I forgot about this one... let me have a look

Reply:
I see. What about only detecting conflicts when both have a non-default value?

It would work if developers set one of the two, of course:

```
@FieldProjection => MAPPING
@FieldProjection(convert = YES) => MAPPING
@FieldProjection(model = MAPPING) => MAPPING
@FieldProjection(convert = NO) => INDEX
@FieldProjection(model = INDEX) => INDEX
@FieldProjection(model = STRING) => STRING
@FieldProjection(model = RAW) => RAW
```

It would correctly fail in most cases when developers explicitly set two to non-default values:

```
@FieldProjection(convert = NO, model = INDEX) => ERROR
@FieldProjection(convert = NO, model = RAW) => ERROR
@FieldProjection(convert = NO, model = STRING) => ERROR
```

It incorrectly ignore the conflict when developers explicitly set one default and one non-default value:

```
@FieldProjection(convert = YES, model = INDEX) => INDEX ... :/
@FieldProjection(convert = YES, model = RAW) => RAW ... :/
@FieldProjection(convert = YES, model = STRING) => STRING ... :/
@FieldProjection(convert = NO, model = MAPPING) => INDEX ... :/
```

But then those last few examples are far-fetched... no?

Reply:
> I just thought ... maybe a bad idea, but what if we add `ValueConvert.DEFAULT` and `ValueModel.DEFAULT` and use these in the annotation, mark both as deprecated, once we remove ValueConvert, we'll remove the `ValueModel.DEFAULT` constant too... that way we will know that if the value is not `DEFAULT` the user set it explicitly ...

+1 even better.

Adding an enum value is technically a breaking change since people might be doing `switch(valueConvert) { case YES:  /* ... */ return; case NO: /* ... */ return; }` somewhere, and will now end up with a compilation error.
But then, we were planning to add `ValueConvert.PARSE` before, and that would have broke their app just the same :shrug: